src/OpenTelemetry.Api/Context/RuntimeContext.cs

```diff
@@ -67,7 +67,14 @@ public static RuntimeContextSlot<T> RegisterSlot<T>(string name)
         /// <returns>The slot previously registered.</returns>
         public static RuntimeContextSlot<T> GetSlot<T>(string name)
         {
-            return (RuntimeContextSlot<T>)Slots[name];
+            if (Slots.TryGetValue(name, out var slot) && slot is RuntimeContextSlot<T> expectedSlot)
+            {
+                return expectedSlot;
+            }
+            else

```

Comment:
nit: either with `TryGetSlot` or `GetSlot` you can get rid of the `else` statement